File: server2/view/json.js
module.exports = function(handler){
	return async function(req,res,next){
		try{
			var result = await handler(req,res,next);
			res.send(JSON.stringify({
				code:0,
				msg:'',
				data:result
			}));
		}catch(e){
			//if( e.stack )
			//	console.log(e.stack);
			res.send(JSON.stringify({
				code:1,
				msg:e.message,
				data:null
			}));
		}
	};
};

File: server3/view/json.go
package view

import (
	"github.com/gin-gonic/gin"
	"net/http"
	"encoding/json"
)

type resultType struct{
	Code int `json:"code"`
	Msg string `json:"msg"`
	Data interface{} `json:"data"`
}

func Json(handler func(*gin.Context)(interface{},error) )(func (*gin.Context)){
	return func (c *gin.Context){
		result := &resultType{};
		handlerResult,error := handler(c);
		if error != nil{
			result.Code = 1;
			result.Msg = error.Error();
		}else{
			result.Code = 0;
			result.Msg = "";
			result.Data = handlerResult;
		}
		resultString,error := json.Marshal(&result);
		if error != nil{
			c.String(http.StatusOK,"{code:1,msg:'json序列化失败',data:''}");
		}else{
			c.String(http.StatusOK,string(resultString));
		}
	}
}